docs/scenarios/jdk_8_to_17_migration.md

```diff
@@ -0,0 +1,428 @@
+   ![Provider settings](../images/jdk_8_to_17_migration/provider_settings.png)
+
+4. Click on `Start Analyzer` to initialize the RPC server
+
+![start_analyzer.png](../images/jdk_8_to_17_migration/start_analyzer.png)
+
+## Step 2: Understanding the Custom Rule
+
+Before we run the analysis, lets understand the custom rule we are going to use.
+
+#### 1. Custom Rule for MD5 Hashing
+
+The `Library.java` class is responsible for managing books in the system. However, it uses MD5 hashing to generate unique identifiers:
+
+Library.java (Before Migration)
+
+```java
+    ...
+    ...
+    private String generateHash(String input) {
+        try {
+            MessageDigest md5Digest = MessageDigest.getInstance("MD5");
+            byte[] hash = md5Digest.digest(input.getBytes());
+            BASE64Encoder encoder = new BASE64Encoder();
+            return encoder.encode(hash);
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+```
+
+##### Problem:
+
+MD5 is outdated and vulnerable to collisions.
+
+##### Solution:
+
+We need to detect MD5 usage and replace it with SHA-256.
+
+Since MD5 hashing is insecure, we add a custom rule to detect and recommend secure alternatives.
+
+````yaml
+- ruleID: java-custom-security-00001
+  category: mandatory
+  effort: 2
+  labels:
+    - konveyor.io/source=java8
+    - konveyor.io/source=openjdk
+    - konveyor.io/target=openjdk17
+  when:
+    and:
+      - java.referenced:
+          pattern: java.security.MessageDigest.getInstance
+          location: METHOD_CALL
+      - builtin.filecontent:
+          pattern: 'MessageDigest.getInstance\("MD5"\)'
+  description: Do not use MD5 for hashing due to security vulnerabilities.
+  message: |
+    The MD5 algorithm is **outdated** and should not be used for secure hashing.
+    Instead, use **SHA-256** or **SHA-512**.
+
+    **Fix:**
+    ```java
+    // Deprecated MD5 usage
+    MessageDigest md = MessageDigest.getInstance("MD5");
+
+    // Recommended SHA-256 usage
+    MessageDigest md = MessageDigest.getInstance("SHA-256");
+    ```
+  links:
+    - title: "java ref"
+      url: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/security/MessageDigest.html
+    - title: "MD5"
+      url: https://csrc.nist.gov/projects/hash-functions
+````
+
+You can learn more about writing your own custom rules [here](https://github.com/konveyor/kai/blob/main/docs/custom_ruleset.md)
+
+## Step 3: Run Analysis
+
+Let's run our initial analysis:
+
+1. Once you have RPC server initialized, navigate to "Konveyor Analysis View" and click Run Analysis. Open the command palette by pressing `Command + Shift + P` to find it.
+   ![run_analysis.png](../images/jdk_8_to_17_migration/run_analysis.png)
```

Comment:
I am getting an incident identified for rule.yaml as well 
@savitharaghunathan . Should I copy the rules somewhere else so that it is excluded or is it a bug 
![image](https://github.com/user-attachments/assets/5e851b2a-9453-4bc2-b760-c2a190520699)

Reply:
@sshveta when you loaded the project into vscode, did you select the `migrationex` subfolder? It should only have access to pom.xml and `src` folder.